backend/business-partner-agent/src/test/java/org/hyperledger/bpa/persistence/repository/PartnerProofRepositoryTest.java

```diff
@@ -80,4 +79,53 @@ void testExchangeStateDecorator() {
                 pp.getStateToTimestamp().toApi().keySet().toArray()[2]);
     }
 
+    @Test
+    void testGetPresentationExchangeListByPartnerId(){
+      PartnerProof pp = createRandomPartnerProof();
+      int numberOfPresExRecords = 42;
+      for(int i = 1; i<numberOfPresExRecords; i++) {
+        repo.save(createDummyPresEx(pp));
+      }
+
+      assertEquals(42, repo.findByPartnerId(
+        pp.getPartnerId(), Pageable.unpaged()).getNumberOfElements());
+
+      assertEquals(9, repo.findByPartnerId(
```

Comment:
You could add a sort param for the presentation exchange id here and then test if you have the right ones in the page e.g. 0 - 5

Reply:
When i try to sort the records by the presentation exchange id, which is a string, i ran into a sort problem, where the records were not sorted in a "natural" order means: "pres-1 , pres-10, pres11... etc." i was looking for an natural sort param in the Sort.class, but i could not find it ad-hoc.  [Sort.Order Docu](https://micronaut-projects.github.io/micronaut-data/1.0.0/api/io/micronaut/data/model/Sort.Order.html)
the following screenshot illustrates the result of the first two entries after sorting by "presentationExchangeId":
![image](https://user-images.githubusercontent.com/104754782/180824947-1a3aea90-9a84-48c3-bab7-fcc2b1238a95.png)

i created tests where i count with that behavior, so the tests pass, but i think thats not intended.